question/bank/history/classes/question_history_view.php

```diff
@@ -0,0 +1,227 @@
+     */
+    public function __construct($contexts, $pageurl, $course, $entryid, $returnurl, $cm = null) {
+        parent::__construct($contexts, $pageurl, $course, $cm);
+        $this->entryid = $entryid;
+        $this->basereturnurl = $returnurl;
+    }
+
+    protected function get_question_bank_plugins(): array {
+        $questionbankclasscolumns = [];
+        $corequestionbankcolumns = [
+                'checkbox_column',
+                'question_type_column',
+                'question_name_idnumber_tags_column',
+                'edit_menu_column',
+                'edit_action_column',
+                'tags_action_column',
+                'preview_action_column',
+                'delete_action_column',
+                'export_xml_action_column',
+                'question_status_column',
+                'version_number_column',
+                'creator_name_column',
+                'modifier_name_column',
+                'comment_count_column'
+        ];
+        if (question_get_display_preference('qbshowtext', 0, PARAM_BOOL, new \moodle_url(''))) {
+            $corequestionbankcolumns[] = 'question_text_row';
+        }
+
+        foreach ($corequestionbankcolumns as $fullname) {
+            $shortname = $fullname;
+            if (class_exists('qbank_history\\' . $fullname)) {
+                $fullname = 'qbank_history\\' . $fullname;
+                $questionbankclasscolumns[$shortname] = new $fullname($this);
+            } else if (class_exists('core_question\\local\\bank\\' . $fullname)) {
+                $fullname = 'core_question\\local\\bank\\' . $fullname;
+                $questionbankclasscolumns[$shortname] = new $fullname($this);
+            } else {
+                $questionbankclasscolumns[$shortname] = '';
+            }
+        }
+        $plugins = \core_component::get_plugin_list_with_class('qbank', 'plugin_feature', 'plugin_feature.php');
+        foreach ($plugins as $componentname => $plugin) {
+            $pluginentrypointobject = new $plugin();
+            $pluginobjects = $pluginentrypointobject->get_question_columns($this);
+            // Don't need the plugins without column objects.
+            if (empty($pluginobjects)) {
+                unset($plugins[$componentname]);
+                continue;
+            }
+            foreach ($pluginobjects as $pluginobject) {
+                $classname = new \ReflectionClass(get_class($pluginobject));
+                foreach ($corequestionbankcolumns as $key => $corequestionbankcolumn) {
+                    if (!\core\plugininfo\qbank::is_plugin_enabled($componentname)) {
+                        unset($questionbankclasscolumns[$classname->getShortName()]);
+                        continue;
+                    }
+                    // Check if it has custom preference selector to view/hide.
+                    if ($pluginobject->has_preference()) {
+                        if (!$pluginobject->get_preference()) {
```

Comment:
Could these two ifs be combined into the one statement?

Reply:
merged the logic